src/edgepi/adc/edgepi_adc.py

```diff
@@ -917,6 +930,11 @@ def set_config(
         """
         # pylint: disable=unused-argument
 
+        if adc_1_analog_in is not None:
+            adc_1_analog_in = self.__analog_in_to_adc_in_map[adc_1_analog_in]
+        if adc_2_analog_in is not None:
+            adc_2_analog_in = self.__analog_in_to_adc_in_map[adc_2_analog_in]
+
```

Comment:
in general it is not a good idea that you are changing the type of the variable at runtime like that.
`adc_1_analog_in` and `adc_2_analog_in` are of type `AnalogIn` but you are changing them into `ADCChannel`.

I would introduce new variables like `adc_1_ch` and `adc_2_ch`. 
also I think you really don't need to use if statements here. you can have something like:
```
            adc_1_ch = self.__analog_in_to_adc_in_map.get(adc_1_analog_in)
            adc_2_ch = self.__analog_in_to_adc_in_map.get(adc_2_analog_in)
```

Reply:
> in general it is not a good idea that you are changing the type of the variable at runtime like that. `adc_1_analog_in` and `adc_2_analog_in` are of type `AnalogIn` but you are changing them into `ADCChannel`.
> 
> I would introduce new variables like `adc_1_ch` and `adc_2_ch`. also I think you really don't need to use if statements here. you can have something like:
> 
> ```
>             adc_1_ch = self.__analog_in_to_adc_in_map.get(adc_1_analog_in)
>             adc_2_ch = self.__analog_in_to_adc_in_map.get(adc_2_analog_in)
> ```

the type of variable changes because we want to filter arguments to pass to __set_config(). If we introduce another set of variables, the filtered arguments will include variables that are not recognized by the __set_config(). The new type is introduced in accordance with #289.

Reply:
why can't we change `__config` like:
```
    def __config(
        self,
        adc_1_ch: CH = None,
        adc_2_ch: CH = None,

```
wouldn't this make more sense? because `__config` is taking `ADCChannel` type not `AnalogIn` type and new variable names match the type better.

You just need to refactor the code to use `adc_1_ch` and `adc_2_ch`. 

I think this refactoring is worth it to avoid future confusion.

Reply:
2ba617587852cd18c2721414a95fd8be2bd7a289